server/src/main/java/org/elasticsearch/common/blobstore/BlobContainer.java

```diff
@@ -144,6 +144,22 @@ void writeMetadataBlob(
         CheckedConsumer<OutputStream, IOException> writer
     ) throws IOException;
 
+    /**
+     * Reads blob content from the input stream and writes it to the container in a new blob with the given name,
+     * using an atomic write operation if the implementation supports it.
+     *
+     * @param purpose             The purpose of the operation
+     * @param blobName            The name of the blob to write the contents of the input stream to.
+     * @param inputStream         The input stream from which to retrieve the bytes to write to the blob.
+     * @param blobSize            The size of the blob to be written, in bytes.  It is implementation dependent whether
+     *                            this value is used in writing the blob to the repository.
```

Comment:
Can we strengthen this to say that it must be the amount of bytes in the input stream? Not asking to verify that (we do for S3 multi-part at least though), but this is hardly a specification of the parameter.

Reply:
I am not sure the new contract will hold for the existing spot (at least I did not check), perhaps we should avoid changing that for now (unless you did a thorough check of it?).

Reply:
We can also see what CI thinks ofc.

Reply:
It should hold for that, as writeBlob (non-atomic) also writes the same amount of bytes, just in a non-atomic fashion (e.g., another thread might read partial results). Lets see what CI thinks.

Reply:
I manually checked our Azure, GCP, S3, FS implementations and they use the right blob-size rather than a different one. So I think the new comment and check hold.